addons/knowledge/static/src/js/widgets/knowledge_emoji_picker.js

```diff
@@ -29,30 +27,116 @@ const EmojiPickerWidget = Widget.extend(WidgetAdapterMixin, {
              * @param {String} unicode
              */
             onClickEmoji: unicode => {
-                this.trigger_up('emoji_click', {
-                    articleId: this.options.article_id,
+                this.__parentedParent.trigger_up('emoji_click', {
+                    articleId: this.articleId || false,
                     unicode: unicode || false,
                 });
                 this.close();
             },
         });
-        const menu = this.el.querySelector('.dropdown-menu');
-        return this.component.mount(menu);
+        return this.component.mount(this.el);
     },
-
     /**
-     * Closes the dropdown
+     * Close (hide) the emojiPicker
      */
     close: function () {
-        this.$('.dropdown-menu').removeClass('show');
+        window.removeEventListener('resize', this.windowChangeHandler);
+        window.removeEventListener('scroll', this.windowChangeHandler);
+        this.el.style.display = 'none';
+    },
+    /**
+     * Show the emojiPicker and move it close to its target. Depending on the
+     * space available, it will be above or below the target, with its width
+     * to the right or to the left. The emojiPicker is displayed along with an
+     * invisible div which will force the user to close the emojiPicker before
+     * doing anything else. Listen to window scroll and resize events to replace
+     * the emojiPicker at a valid position.
+     *
+     * @param {Object} options
+     * @param {[HTMLElement]} currentTarget element which need an emoji change
+     *                        false => use the previous currentTarget
+     * @param {[boolean]} withTransition the emojiPicker will be moved to its
+     *                    target destination with an animation if True.
+     */
+    showOnTarget: function ({currentTarget=false, withTransition=false}) {
```

Comment:
Why an object-like definition ? Why not 2 parameters (as those are not really "options") ?

Reply:
't was to use the function with only the required parameter:
`showOnTarget({currentTarget: tagada})`
and
`showOnTarget({withTransaction: tagada})`